sentry/src/main/java/io/sentry/SpanId.java

```diff
@@ -0,0 +1,35 @@
+package io.sentry;
+
+import io.sentry.util.Objects;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+
+public final class SpanId {
+  private final @NotNull String value;
+
+  public SpanId(final @NotNull String value) {
+    this.value = Objects.requireNonNull(value, "value is required");
+  }
+
+  public SpanId() {
+    this(UUID.randomUUID().toString().replace("-", "").substring(0, 16));
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    SpanId spanId = (SpanId) o;
+    return value.equals(spanId.value);
+  }
+
+  @Override
+  public int hashCode() {
+    return value.hashCode();
+  }
```

Comment:
do we need this for testing only? if so, could we compare the `value` instead?

Reply:
I added it because `SpanId` is a value object so I think it's good to add equals and hashcode by default?

Reply:
yes, it makes sense, `SentryId` also has it, thx